### PR TITLE
NMS-16377: Add raw data to flow document

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://opennms.atlassian.net/browse/NMS-16377

### CODE REVIEW

File: features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/transport/Netflow9MessageBuilder.java [hunk 1]
```diff
@@ -39,6 +39,7 @@
 import static org.opennms.netmgt.telemetry.protocols.netflow.parser.transport.MessageUtils.setLongValue;
 
 import java.net.InetAddress;
+import java.util.stream.Collectors;
```
Comment (fooker) [inline features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/transport/Netflow9MessageBuilder.java:42]: Revert

File: features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/transport/Netflow5MessageBuilder.java [hunk 2]
```diff
@@ -47,6 +48,8 @@
 import org.opennms.netmgt.telemetry.protocols.netflow.transport.NetflowVersion;
 import org.opennms.netmgt.telemetry.protocols.netflow.transport.SamplingAlgorithm;
 
+import com.google.common.collect.Streams;
```
Comment (fooker) [inline features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/transport/Netflow5MessageBuilder.java:51]: Revert

File: features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/transport/IpFixMessageBuilder.java [hunk 3]
```diff
@@ -40,17 +40,42 @@
 
 import java.net.InetAddress;
 import java.time.Instant;
+import java.util.List;
```
Comment (fooker) [inline features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/transport/IpFixMessageBuilder.java:43]: Revert

File: features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/ParserBase.java [hunk 4]
```diff
@@ -470,4 +475,27 @@ private List<String> correctFlow(final FlowMessage.Builder flow) {
...
+        if (!includeRawMessage) {
+            return flowMessageBuilder;
+        }
+
+        flowMessageBuilder.addAllRawMessage(Streams.stream(values)
+                .map(value -> {
+                    final TransformationVisitor transformationVisitor = new TransformationVisitor();
+                    value.visit(transformationVisitor);
+                    return transformationVisitor.getValue();
+                }).collect(Collectors.toList()));
```
Comment (fooker) [inline features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/ParserBase.java:497]:
```suggestion
        for (final var value : values) {
            final TransformationVisitor transformationVisitor = new TransformationVisitor();
            value.visit(transformationVisitor);
            flowMessageBuilder.addRawMessage(transformationVisitor.getValue());
        }
```

File: features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/ParserBase.java [hunk 5]
```diff
@@ -328,7 +333,7 @@ protected CompletableFuture<?> transmit(final RecordProvider packet, final Sessi
                         // Let's serialize
                         final FlowMessage.Builder flowMessage;
                         try {
-                            flowMessage = this.getMessageBuilder().buildMessage(record, enrichment);
+                            flowMessage = transformRawMessage(includeRawMessage, this.getMessageBuilder().buildMessage(record, enrichment), record);
```
Comment (fooker) [inline features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/ParserBase.java:336]: You can just inline it, work on the builder directly and maybe even move this out of the try-catch-block?

File: features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/transport/TransformationVisitor.java [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+import org.opennms.netmgt.telemetry.protocols.netflow.parser.ie.values.OctetArrayValue;
+import org.opennms.netmgt.telemetry.protocols.netflow.parser.ie.values.SignedValue;
+import org.opennms.netmgt.telemetry.protocols.netflow.parser.ie.values.StringValue;
+import org.opennms.netmgt.telemetry.protocols.netflow.parser.ie.values.UndeclaredValue;
+import org.opennms.netmgt.telemetry.protocols.netflow.parser.ie.values.UnsignedValue;
+import org.opennms.netmgt.telemetry.protocols.netflow.transport.Semantic;
+
+import com.google.protobuf.ByteString;
+
+public class TransformationVisitor implements Value.Visitor {
```
Comment (fooker) [inline features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/transport/TransformationVisitor.java:51]: Naming: `TransportValueVisitor`?

**Review by fooker — CHANGES_REQUESTED** [review]

File: features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/transport/Netflow5MessageBuilder.java (hunk 2 above)
Comment (christianpape) [reply features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/transport/Netflow5MessageBuilder.java]: done

File: features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/transport/IpFixMessageBuilder.java (hunk 3 above)
Comment (christianpape) [reply features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/transport/IpFixMessageBuilder.java]: done

File: features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/ParserBase.java (hunk 4 above)
Comment (christianpape) [reply features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/ParserBase.java]: done

File: features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/ParserBase.java (hunk 5 above)
Comment (christianpape) [reply features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/ParserBase.java]: done

**Review by fooker — APPROVED** [review]

**Pull request merged**